### DISCUSSION TITLE
[PSD4] Esercizio Slide 6

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#137] [Esercizi (A-L)]
**Esercizio (1+2+1 punti)**

Dati i numeri in rappresentazione IEEE 754 X = 0x13600000 e Y = 0x41100000, 
(a) rappresentarli in notazione decimale in virgola mobile (approssimato +/- 0,03)
(b) eseguire l’operazione X+Y
(c) rappresentare il risultato sia in notazione decimale a virgola mobile e sia in esadecimale (approssimato +/- 0,03)

**Comment by FeddyLix17** [comment]
0x41100000<sub>16</sub> = 0 10000010 00100000000000000000000 <sub>IEEE 754</sub> = 1,001 * 2<sup>130 - 127 = 3</sup> = 1001<sub>2</sub> = 9<sub>10</sub>
il numero 0x13600000 l'ho calcolato usando https://babbage.cs.qc.cuny.edu/ieee-754.old/32bit.html
0x13600000<sub>16</sub> = 0 00100110 11000000000000000000000<sub>IEEE 754</sub> = 1,11 * 2<sup>38 - 127 = -89</sup> = 0.00...111<sub>2</sub> = 2.8272775 * 10<sup>-27</sup><sub>10</sub>
Stando a photomat e chatgpt essendo 0x13600000<sub>16</sub> un numero veramente piccolo la somma con 0x41100000<sub>16</sub> è praticamente trascurabile, di conseguenza 0x41100000<sub>16</sub> (9<sub>10</sub>) +  0x13600000<sub>16</sub> ≈  9<sub>10</sub>
La rappresentazione del risultato sarà quindi la medesima del numero 0x41100000<sub>16</sub>
